src/utils/components/TextInput.vue

```diff
@@ -0,0 +1,159 @@
+<script>
+// Global variable to count the components created in order to create a unique DOM id.
+let components = 0
+</script>
+<script setup>
+/** Input with clear button component */
+import { computed, nextTick, ref, toRefs, watch } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+// On each component creation set the current component ID and increase the counter
+const clearButtonId = `button-addon-clear-${components}`
+components = components + 1
```

Comment:
you might want to do that on the `onMounted` hook, I'm not sure this will be called many times if you instantiate more than one component with a single `import` statement 🤔

Reply:
That's the idea and hack here, this ID is used as HTML ID which should be unique in the DOM if I'm correct, so with this hack each instance of `TextInput` has a different ID.